pkgs/leak_tracker_flutter_testing/lib/src/matchers.dart

```diff
@@ -5,53 +5,50 @@
 import 'package:flutter/foundation.dart';
 import 'package:flutter_test/flutter_test.dart';
 
-/// Checks if the object dispatches events to `MemoryAllocations.instance`.
-///
-/// The memory events are used by tools like leak_tracker for diagnostics.
-///
-/// The matcher checks that the object object is instrumented properly,
-/// dispatches two events to `MemoryAllocations.instance`,
-/// first `ObjectCreated` and then `ObjectDisposed`.
-Matcher dispatchesMemoryEvents(Type type) {
-  return _DispatchesMemoryEvents(type);
+/// Invokes [callback] and collects events dispatched to [MemoryAllocations.instance] for [type].
+Future<List<ObjectEvent>> memoryEvents(
+  Future<void> Function() callback,
+  Type type,
+) async {
+  final events = <ObjectEvent>[];
+
+  void listener(ObjectEvent event) {
+    if (event.object.runtimeType == type) {
+      events.add(event);
+    }
+  }
+
+  MemoryAllocations.instance.addListener(listener);
+  await callback();
+  MemoryAllocations.instance.removeListener(listener);
+
+  return events;
 }
 
-class _DispatchesMemoryEvents extends Matcher {
-  const _DispatchesMemoryEvents(this.type);
+/// Checks if Iterable<ObjectEvent> contains two events, first `ObjectCreated` and then `ObjectDisposed`.
+Matcher areCreateAndDispose = const _AreCreateAndDispose();
```

Comment:
Nit: consider naming this `isCreatedAndDisposed` and the actual `Matcher` `_IsCreatedAndDisposed`.

Reply:
The validated item is changed now. Now it is list of events. And matcher verifies that events 'are create and dispose'.